scripts/tests/chiptest/__init__.py

```diff
@@ -17,34 +17,38 @@
 from pathlib import Path
 import os
 import logging
+import subprocess
+import re
 
 import chiptest.linux
 import chiptest.runner
 
 from .test_definition import TestTarget, TestDefinition, ApplicationPaths
 
 
-def AllTests(root: str):
-    """Gets all the tests that can be found in the ROOT directory based on
-       yaml file names.
+def AllTests(chip_tool: str):
+    """Executes `chip_tool` binary to see what tests are available.
     """
-    for path in Path(os.path.join(root, 'src', 'app',  'tests', 'suites')).rglob("*.yaml"):
-        logging.debug('Found YAML: %s' % path)
 
-        # grab the name without the extension
-        name = path.stem
+    result = subprocess.run([chip_tool, 'tests'], capture_output=True)
 
-        if 'Simulated' in name:
+    # Unfortunately there is no 'LIST' command and result return code is 1 (i.e. error)
+    # however the above will return tests in stderr
```

Comment:
It should be pretty straightforward to add a list command if that makes your life simpler ?

Reply:
I just made #12322 for that.

Reply:
done